packages/keyda-demo/src/lib/Context/index.js

```diff
@@ -3,30 +3,62 @@ import React, { createContext, useContext, useReducer } from 'react';
 const initialKeyState = {
   keyTimeList: [],
   userId: '',
+  trainCount: 0,
+  inputRef: {},
 };
 
 const keyStateReducer = (state, action) => {
-  const { keyTimeList, userId } = state;
+  const { keyTimeList, userId, trainCount, inputRef } = state;
   switch (action.type) {
+    case 'SET_REF':
+      console.log(state, action);
+      return {
+        keyTimeList: keyTimeList,
+        userId: userId,
+        trainCount: trainCount,
+        inputRef: action.inputRef,
+      };
     case 'KEY_DOWN':
       return {
         keyTimeList: [...keyTimeList, ...action.newKeyTime],
-        userId: state.userId,
+        userId: userId,
+        trainCount: trainCount,
+        inputRef: inputRef,
       };
     case 'KEY_UP':
       return {
         keyTimeList: [...keyTimeList, ...action.newKeyTime],
-        userId: state.userId,
+        userId: userId,
+        trainCount: trainCount,
+        inputRef: inputRef,
       };
     case 'TYPE_USER_ID':
       return {
-        keyTimeList: state.keyTimeList,
+        keyTimeList: keyTimeList,
         userId: userId === action.userId ? userId : action.userId,
+        trainCount: trainCount,
+        inputRef: inputRef,
       };
     case 'BACKSPACE':
       return {
         keyTimeList: [],
         userId: userId,
+        trainCount: trainCount,
+        inputRef: inputRef,
+      };
+    case 'REGISTER':
+      return {
+        keyTimeList: [],
+        userId: '',
+        trainCount: trainCount < 5 ? trainCount + 1 : trainCount,
+        inputRef: inputRef,
+      };
+    case 'SUBMIT':
+      return {
+        keyTimeList: [],
+        userId: '',
+        trainCount: 0,
+        inputRef: {},
       };
```

Comment:
Added new action type for `REGISTER` to count submission and `SUBMIT`to reset all the states after the whole submission process is done.